.buildkite/scripts/resolve_es_branch.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+set -e
+
+VERSION_URL="https://raw.githubusercontent.com/elastic/logstash/main/ci/logstash_releases.json"
+
+echo "Fetching versions from $VERSION_URL"
+VERSIONS=$(curl --silent $VERSION_URL)
```

Comment:
I usually like to protect CI jobs from transient network failures. There are two avenues here:

1. Use curl's native `--retry`/`--retry-delay` e.g. `VERSIONS=$(curl --retry 5 --retry-delay 5 -fsSL $VERSION_URL)` (`S` -> show error messages if request fails, in spite of `s` -> silent ; `L` follow redirects, `f` returns an error (thus ending the script if the HTTP response status in an error code)

2. Alternatively, if instead of `family/core-ubuntu-2204` you use one of the custom logstash CI vm images like `family/platform-ingest-logstash-ubuntu-2204`, you can take advantage of the pre-installed `retry` command with something like `VERSIONS=$(retry -t 5 -- curl -fsSL $VERSION_URL)`.
3.

Reply:
Thanks so much for great feedback.
I have addressed and resolved _obvious_ changes.

- as we are using `family/core-ubuntu-2204` VM image (as discussed before need to spin up docker in CI), I have preferred 2nd option of suggestions;

Reply:
> Thanks so much for great feedback. I have addressed and resolved _obvious_ changes.
> 
> * as we are using `family/core-ubuntu-2204` VM image (as discussed before need to spin up docker in CI), I have preferred 2nd option of suggestions;

You are welcome! Seems to be based on the [latest commit](https://github.com/elastic/logstash-filter-elastic_integration/pull/103/commits/afa65dc41ea10e4678fa2aaf1361f585eeee8a52) that you actually went with option 1. No problems at all there, just wanted to make sure we are on the same page.